test/functional/collection.test.js

```diff
@@ -1313,33 +1313,41 @@ describe('Collection', function() {
     }
   });
 
-  it('should createIndex with serverSelection readPreference primary', {
-    metadata: {
-      requires: { topology: 'replicaset' }
-    },
-    test: function(done) {
-      const configuration = this.configuration;
-      const client = configuration.newClient(configuration.writeConcernMax(), {
-        useUnifiedTopology: true,
-        readPreference: 'primaryPreferred'
-      });
-      client.connect((err, client) => {
-        expect(err).to.not.exist;
-        const db = client.db(configuration.db);
-        const collection = db.collection('db-two');
-        TopologyStub = sinon.spy(Topology.prototype, 'selectServer');
-        collection.createIndex({ quote: 'text' }, (err, res) => {
-          expect(err).to.not.exist;
-          expect(res).to.equal('quote_text');
-          expect(TopologyStub.called).to.equal(true);
-          expect(TopologyStub.args[0][0].readPreference.mode).to.equal(ReadPreference.PRIMARY);
-          client.close(done);
-        });
-      });
-    }
-  });
+  context.only('DLL methods with serverSelection readPreference primary', () => {
+    const primaryReadPreferenceDLL = {
```

Comment:
@mbroadst is this what you mean by data-driven?

Reply:
yep!